MapboxNavigation/NavigationViewController.swift

```diff
@@ -207,8 +207,10 @@ open class NavigationViewController: UIViewController, NavigationStatusPresenter
         
         self.navigationService = options?.navigationService ?? MapboxNavigationService(route: route)
         self.navigationService.delegate = self
-        self.voiceController = options?.voiceController ?? MapboxVoiceController(navigationService: navigationService, speechClient: SpeechSynthesizer(accessToken: navigationService?.directions.accessToken, host: navigationService?.directions.apiEndpoint.host))
+//        self.voiceController = options?.voiceController ?? MapboxVoiceController(navigationService: navigationService, speechClient: SpeechSynthesizer(accessToken: navigationService?.directions.accessToken, host: navigationService?.directions.apiEndpoint.host))
 
+        self.voiceController = options?.voiceController ?? RouteVoiceController(navigationService: navigationService)
```

Comment:
`NavigationOptions.voiceController` was intended as a customization hook to directly swap out speech synthesizers. Now RouteVoiceController is an extra layer of indirection: the developer would create a MySpeechSynthesizer to pass into the RouteVoiceController initializer to pass into the NavigationOptions initializer to pass into the NavigationViewController initializer. But RouteVoiceController itself doesn’t seem to do anything that the developer would need to override, so how about we replace `NavigationOptions.voiceController` with `NavigationOptions.speechSynthesizers` and plumb that value through?

Reply:
If we replace `NavigationOptions.voiceController` with `NavigationOptions.speechSynthesizer` we'll loose ability to provide custom subclasses of `RouteVoiceController`. To preserve it and to enable easy speech synthesizer customization we'll have to invent something else, which will also correctly handle custom speech synth + route voice controller simultaneously. 
I couldn't come up with a brilliant idea right away, but maybe we can just make `RouteVoiceController.speechSynthesizer` property writable so user can easily customize speech synthesizing during initialization? This case might require some extensive testing for speech synthesizers do correctly swap mid-navigation.